pkg/controller/job/job_controller.go

```diff
@@ -812,9 +813,7 @@ func (jm *Controller) manageJob(job *batch.Job, activePods []*v1.Pod, succeeded
 		klog.V(4).InfoS("Too many pods running for job", "job", klog.KObj(job), "deleted", len(podsToDelete), "target", parallelism)
 		removed, err := jm.deleteJobPods(job, jobKey, podsToDelete)
 		active -= removed
-		if err != nil {
-			return active, err
-		}
+		return active, metrics.JobSyncActionPodsDeleted, err
```

Comment:
add a comment indicating that we either delete or add pods in a cycle, and deleting pods takes precedence.

Reply:
this wasn't addressed

Reply:
ah, must have missed it. done.